### PR TITLE
add cf and spectre flags

### PR DESCRIPTION
**Pull request by TylerLeonhardt**

### CODE REVIEW

**Comment by Tyriar** [comment]
What's up with this change? We are currently not on the latest version of node-pty due to not resolving issues that arose in vscode when updating to https://github.com/microsoft/node-pty/pull/487

File: binding.gyp [hunk 1]
```diff
@@ -13,7 +13,21 @@
           ],
           'libraries': [
             'shlwapi.lib'
-          ]
+          ],
+          'msvs_settings': {
+            'VCCLCompilerTool': {
+              'ExceptionHandling': 1,
```
Comment (deepak1556) [inline binding.gyp:19]: `ExceptionHandling` is unrelated to this change, should be removed.

Comment (TylerLeonhardt) [reply binding.gyp]: removed

**Review by Tyriar — COMMENTED** [review]
Deferring approval to @deepak1556

**Review by sbatten — APPROVED** [review]

**Review by deepak1556 — APPROVED** [review]
Pushed a change to remove redundant declarations. `target_defaults` will apply to all `targets` in the file.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -54,7 +54,7 @@ jobs:
 
 - job: Windows
   pool:
-    vmImage: 'vs2017-win2016'
+    vmImage: 'windows-latest'
```
Comment (deepak1556) [inline azure-pipelines.yml:57]: @Tyriar I have moved these from deprecated images.

**Comment by deepak1556** [comment]
Module tests does not pass with `windows-2022` agent https://github.com/microsoft/node-pty/runs/5583465216, this might be further investigation for future agent updates. Reverted to `windows-2019` agent.

**Comment by deepak1556** [comment]
Sorry, yeah this PR is good to go, don't have plans to look into the `windows-2022` agent failure for now since vscode core still uses 2019 agent. Happy if others want to tackle it now.

**Pull request merged**